pkg/kubelet/cm/memorymanager/policy_single_numa.go

```diff
@@ -40,12 +40,12 @@ type systemReservedMemory map[int]map[v1.ResourceName]uint64
 // SingleNUMAPolicy is implementation of the policy interface for the single NUMA policy
 type singleNUMAPolicy struct {
 	// TODO: define new kubelet flag and pass it to the ContainerManager -> MemoryManager -> singleNUMAPolicy
-	// multipleNUMAGroups defines groups of NUMA nodes that will be set to preferred under the topology hint generation
+	// multiNUMAGroups defines groups of NUMA nodes that will be set to preferred under the topology hint generation
 	// for example on the machine with four NUMA nodes and with this parameter equals to [[0,2]], only hints
 	// [0010, 1000, 0101] will be preferred
 	// if this parameter is empty, the default behavior will be used to set preferred hints, it means that only hint
 	// with a minimal amount of nodes that should satisfy the request will be set to preferred
```

Comment:
+1 for description

I may be getting lost with that preferred. IIUC hint should be marked as `preferred` when it is the smallest possible quantity of nodes (based on amount of `allocatable` and not amount of free). 

So hint `[0101]` can be marked as `preferred` only if amount of memory in `allocatable` in each particular NUMA node is not enough for request. (.i.e. request>numa0.allocatable but request<numa0.allocatable+numa1.allocatable)

My suggestion:

> //multiNUMAGroups defines groups of NUMA nodes that will be treated as a joint pool of memory
> // for example on the machine with four NUMA nodes and with this parameter equals to [[0,2]], only hints
> // [0010, 1000, 0101] will be generated, where hints [0010,1000] are marked as preferred
> // if requested amount of memory can't be satisfied by theoretical capacity of each particular NUMA node but can be satisfied by multiNUMAGroup, then hint [0101] will be set as preferred
> // if this parameter is empty, each NUMA node is treated as separate entity